src/main/java/com/example/demo/services/LeaseManagementDropBoxImpl.java

```diff
@@ -100,4 +112,32 @@ public void dropboxCallback(Long leaseId) throws ApiException, EmptyResultDataAc
         //todo check resonse object
         log.info("lease status updated for leaseId: {}", leaseId);
     }
+
+    ZonedDateTime parseZonedDateTime(String dateStr) {
```

Comment:
Maybe to complex for an mvp, but I think you have to take the timezone into account for the start point of the contract. In new york we are 4 or 5 hours behind UTC, which means the contract will start a day earlier than what is listed in the frontend.